functions/slaViolationChecker/src/app/lib/dynamoDB.js

```diff
@@ -11,100 +11,125 @@ const client = new DynamoDBClient({
- */
-exports.findActivityEnd = async (iun, id, type) => {
-  const tableName = "pn-Timelines";
-
-  // 1. get IUN directly and build timelineElementId from event id
-  const params = {
-    TableName: tableName,
+exports.closingElementIdFromIDAndType = (id, type) => {
+  const returnCouple = {
+    mainTimelineElementId: null,
+    alternativeTimelineElementId: null,
   };
-  let altSortKey = null;
 
-  // instead of performing a query with filter, we can construct the partition and the sort key and perform a GetItem (eventually two)
   switch (type) {
     case "VALIDATION":
       // type VALIDATION:
-      // - INSERT in pn-Timelines di un record con category REQUEST_ACCEPTED indica la terminazione di una “attività di validazione”
-      // - INSERT in pn-Timelines di un record con category REQUEST_REFUSED indica la terminazione di una “attività di validazione”:
-      const timelineBaseValidation = id.split("##")[1];
-      const timeLineIdAccepted = timelineBaseValidation + "_request_accepted";
-      const timeLineIdRefused = timelineBaseValidation + "_request_refused";
-      params.Key = {
-        iun: iun,
-        timelineElementId: timeLineIdAccepted,
-      };
-      altSortKey = timeLineIdRefused;
+      // - INSERT in pn-Timelines of a record with category REQUEST_ACCEPTED: VALIDATION activity end
+      // - INSERT in pn-Timelines of a record with category REQUEST_REFUSED: VALIDATION activity end
+      const timelineBaseValidation = id.split("##")[1]; // IUN remains
+      const timeLineIdAccepted =
+        "request_accepted#IUN_" + timelineBaseValidation;
+      const timeLineIdRefused = "request_refused#IUN_" + timelineBaseValidation;
+      returnCouple.mainTimelineElementId = timeLineIdAccepted;
+      returnCouple.alternativeTimelineElementId = timeLineIdRefused;
       break;
     case "REFINEMENT":
       // type REFINEMENT:
-      // - INSERT in pn-Timelines di un record con category REFINEMENT indica la fine di una “attività di consegna” per uno dei destinatari della notifica
-      // - INSERT in pn-Timelines di un record con category NOTIFICATION_VIEWED indica la fine di una “attività di consegna” per uno dei destinatari della notifica
-      const timelineBaseRefinement = id.replace("01_REFIN##", "");
-      const timeLineIdRefinement = timelineBaseRefinement.replace(
-        "##",
-        "_refinement_"
-      );
-      const timeLineIdNotificationViewed = timelineBaseRefinement.replace(
-        "##",
-        "_notification_viewed_"
-      );
-      params.Key = {
-        iun: iun,
-        timelineElementId: timeLineIdRefinement,
-      };
-      altSortKey = timeLineIdNotificationViewed;
+      // - INSERT in pn-Timelines of a record with category REFINEMENT: DELIVERY activity end for one of the recipients
+      // - INSERT in pn-Timelines of a record with category NOTIFICATION_VIEWED: DELIVERY activity end for one of the recipients
+      const timelineBaseRefinement = id
```

Comment:
@FeliceMente Potresti inserire in ogni "CASE" un esempio di ID nei commenti così da chiarire meglio il tipo di replace che viene fatto?

Reply:
non l'avevo fatto perché gli esempi erano nei test, ma ha senso inserirli anche direttamente qui come commenti